### DISCUSSION TITLE
Plotly3d.js

### DISCUSSION THREAD

**Discussion by gganapat** [WLJSTeam/wljs-notebook#234] [Ideas]
The implementation of Plotly.js has been very effective in making 2d plots with features that don't exist (cleanly) for Wolfram plots in WLJS (things like dynamic tooltips for example). Is there any plan to develop similar wrappers for Plotly3d.js?

gani -

**Comment by JerryI** [comment]
Actually current implementation does support 3d, I believe. 

for example https://plotly.com/javascript/3d-scatter-plots/

will work using Plotly wrapper

try this one

```
data = <|
	"z" -> Table[x y, {x,-10,10}, {y,-10,10}],
	"type" -> "surface"
|>;

Plotly[data]
```

PS: I couldn’t find mentioned plotly3d.js :)

**Comment by gganapat** [comment]
Yes, it worked just fine with the approach you have! Here is a more complex figure with Plotly reproduced from the plotly documentation.

<img width="877" alt="Screenshot 2024-07-14 at 7 15 15 PM" src="https://github.com/user-attachments/assets/3b0d2587-dde8-473b-bdf3-6ea1557735e6">

Thanks, gani--

**Reply by JerryI** [reply]
It looks really great, I haven’t explored 3d features of plotly. May be one could also integrate matplotlib, but I am not sure if there is a Javascript version:D

**Comment by gganapat** [comment]
Thanks!
